src/main/java/io/stargate/sgv2/jsonapi/service/shredding/NamedValueContainer.java

```diff
@@ -3,21 +3,41 @@
 import io.stargate.sgv2.jsonapi.util.PrettyPrintable;
 import io.stargate.sgv2.jsonapi.util.PrettyToStringBuilder;
 import java.util.Collection;
-import java.util.Map;
+import java.util.LinkedHashMap;
 import java.util.Objects;
 
 /**
- * Base for all containers that hold {@link NamedValue}s.
+ * Base implementation for a {@link NamedValueContainer} that maintains the order the named values
+ * were added.
  *
- * <p>Useful so we can deal with a container of {@link NamedValue}s in a generic way, provides some
- * helper methods, and implements {@link PrettyPrintable}.
+ * <p>Marked abstract to force the instantiation of a concrete types the define the {@link
+ * NamedValue} type.
+ *
+ * <p>Provides some helper methods and supports {@link PrettyPrintable} to make it easier to debug.
  *
  * @param <NameT> The type of the name, this is the key in the map
  * @param <ValueT> The type of the value stored in the {@link NamedValue}
  * @param <NvT> The type of the {@link NamedValue} stored in the map
  */
-public interface NamedValueContainer<NameT, ValueT, NvT extends NamedValue<NameT, ValueT>>
-    extends Map<NameT, NvT>, PrettyPrintable {
+public abstract class NamedValueContainer<NameT, ValueT, NvT extends NamedValue<NameT, ValueT>>
+    extends LinkedHashMap<NameT, NvT> implements PrettyPrintable {
+
+  public NamedValueContainer() {
+    super();
+  }
+
+  public NamedValueContainer(int initialCapacity) {
+    super(initialCapacity);
+  }
+
+  public NamedValueContainer(NamedValueContainer<NameT, ValueT, NvT> container) {
+    super(container);
+  }
+
+  public NamedValueContainer(Collection<NvT> values) {
+    super();
```

Comment:
(same as earlier -- assuming `super(values);` wouldn't work due to... type mismatch or something?)